libftl/handshake.c

```diff
@@ -114,10 +114,43 @@ ftl_status_t _init_control_connection(ftl_stream_configuration_private_t *ftl) {
   }
 
   ftl->ingest_socket = sock;
-  
+
   return FTL_SUCCESS;
 }
 
+ftl_status_t _test_stream_key_internal(ftl_stream_configuration_private_t *ftl)
```

Comment:
You dont want to use the connect api as it spawns a bunch of threads when it initiates the connection. All i want to do is check if the key i have is valid. 
Essentially, I am caching the stream key and instead of fetching it everytime using http calls I want to check if the key is still valid and if it is, skip the http calls and use it. 
This api is significantly faster than connect and disconnect.